ext/mbstring/php_mbregex.c

```diff
@@ -1519,9 +1528,13 @@ PHP_FUNCTION(mb_ereg_search_init)
 		RETVAL_TRUE;
 	} else {
 		MBREX(search_pos) = ZSTR_LEN(arg_str);
-		RETVAL_FALSE;
+		zend_argument_value_error(1, "must be a valid string in '%s'", php_mb_regex_get_mbctype());
+		RETURN_THROWS();
 	}
 
+	MBREX(search_pos) = 0;
+	RETVAL_TRUE;
```

Comment:
This code is now duplicated with the code in the `if`. Though probably a moot point as this shouldn't throw either.